app/src/components/fineTunes/FineTuneContentTabs/General/InferenceCodeTabs/CopiableCodeTabs.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from "react";
+import {
+  Tabs,
+  TabList,
+  Tab,
+  TabPanels,
+  TabPanel,
+  HStack,
+  IconButton,
+  useToast,
+  Text,
+  Tooltip,
+} from "@chakra-ui/react";
+import SyntaxHighlighter from "react-syntax-highlighter";
+import { tomorrowNightBright } from "react-syntax-highlighter/dist/cjs/styles/hljs";
+import { CopyIcon } from "lucide-react";
+
+export type CodeTab = {
+  title: string;
+  code: string;
+  language: string;
+};
+
+const CopiableCodeTabs = ({ tabs }: { tabs: CodeTab[] }) => {
+  const [tabIndex, setTabIndex] = useState(0);
+
+  const toast = useToast();
+
+  const copyToClipboard = async (text?: string) => {
```

Comment:
We use this in enough places that we should probably extract it into a component that takes the content to be copied as a prop and renders the copy icon.

Reply:
We only use this copy+toast functionality in two places, and the styles are different between those two. I'm happy to create a shared component that accepts different styles, but I'm not sure that it's a win from a code-complexity perspective.

Reply:
I experimented with putting both copy icons in the same component, but styling with a tooltip became too complex, so I decided to just put `copyToClipboard` in its own hook.

Reply:
I see it in `FormattedJson`, `CopiableCode` and here? If the formatting is different enough then that's fair, can make it a hook instead of a component.

Reply:
ah just saw your comment, sg!